src/Templates/Microsoft.CodeAnalysis.AnalyzerTemplate/MyProject.Analyzer/MyProject.Analyzer.csproj

```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <IsPackable>false</IsPackable>
+
+    <!-- Avoid ID conflicts with the package project. -->
+    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
```

Comment:
There is a chance this will cause restore to fail on Linux where the path separator is the same as the package version separator.